client/src/components/WishlistContext.tsx

```diff
@@ -0,0 +1,106 @@
+import { createContext, useState, useEffect } from 'react';
+import { useUser } from './useUser';
+import { type Product } from '../pages/Catalog';
+
+export type WishlistContextValues = {
+  wishlist: Product[];
+  isInWishlist: (productId: number) => boolean;
+  addToWishlist: (product: Product) => void;
+  removeFromWishlist: (productId: number) => void;
+  error?: unknown;
+  isLoading?: boolean;
+};
+
+export const WishlistContext = createContext<WishlistContextValues>({
+  wishlist: [],
+  isInWishlist: () => false,
+  addToWishlist: () => undefined,
+  removeFromWishlist: () => undefined,
+});
+
+export function WishlistProvider({ children }) {
```

Comment:
Thanks for teaching me how to do this! I have changed all the exercises that use context now to do it this way!